superset/views/core.py

```diff
@@ -1122,10 +1123,10 @@ def generate_json(self, datasource_type, datasource_id, form_data,
     @expose('/slice_json/<slice_id>')
     def slice_json(self, slice_id):
         try:
-            viz_obj = self.get_viz(slice_id)
-            datasource_type = viz_obj.datasource.type
-            datasource_id = viz_obj.datasource.id
-            form_data, slc = self.get_form_data()
+            form_data, slc = self.get_form_data(slice_id)
```

Comment:
Does the overriding, e.g. `since` and `until` still work?

Reply:
yeah I talked to @graceguo-supercat about `get_form_data` and it's supposed to encapsulate all of the logic for form data + overrides. you can see [on this line](https://github.com/apache/incubator-superset/blob/2bc089ef8dcec5d98a43cb2ec80aacf1cdc706cf/superset/views/core.py#L1004) that it will override the saved slice params with any in the request

Reply:
Sweet

Reply:
get_form_data doesn't need `slice_id` parameter